### PR TITLE
CIP-18: Standardised Gas and Pricing Estimation Interface

### PR DESCRIPTION
**Pull request by cmwaters**
## Overview

I have submitted this as part of the `Interface` category, but it may be better suited as a CRC. Let me know and I can easily change it

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: cips/cip-tx-interface.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+---
+title: Standardised Gas and Pricing Estimation Interface
+description: A standardised interface for estimating gas usage and gas pricing for transactions
+author: Callum Waters (@cmwaters)
+discussions-to: URL
+status: Draft
+type: Standards Track
+category: Interface
+created: 2025-03-12
```
Comment (rootulp) [inline cips/cip-tx-interface.md:9]:
```suggestion
created: 2024-03-12
```

File: cips/cip-tx-interface.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+## Backwards Compatibility
+
+As this is a new interface, no consideration for backwards compatibility is necessary
+
+## Reference Implementation
+
+The effectiveness of a "standardised" interface depends on the willingness of current and future clients to adopt it as well as the willingness of teams to provide those services. To set a sufficient precendent, both the Node API within `celestia-node` and the consensus node within `celestia-app` will implement client and server implementations respectively, creating an interface between the existing line of communication. That way by default, light nodes will use that API with the trusted provider they are already using for transaction submission.
+
+The consensus node will use the SimulateTx method to estimate the gas used and use the `min_gas_price` parameter within state as the `estimated_gas_price`
```
Comment (rootulp) [inline cips/cip-tx-interface.md:80]: I think `SimulateTx` is often used in conjunction with 

```
      --gas-adjustment float     adjustment factor to be multiplied against the estimate returned by the tx simulation; if the gas limit is set manually this flag is ignored  (default 1)
```

so should the initial implementation return the direct output of `SimulateTx` or multiplied by some factor (i.e. `1.1` or `1.2`)?

File: cips/cip-tx-interface.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
+---
+title: Standardised Gas and Pricing Estimation Interface
+description: A standardised interface for estimating gas usage and gas pricing for transactions
+author: Callum Waters (@cmwaters)
+discussions-to: URL
```
Comment (rootulp) [inline cips/cip-tx-interface.md:5]: Note: this needs a URL pre-merge.

**Review by rootulp — COMMENTED** [review]
Overall LGTM! Really well written

File: cips/cip-tx-interface.md (hunk 3 above)
Comment (cmwaters) [reply cips/cip-tx-interface.md:5]: Yeah I haven't yet created the forum post but will do that today

File: cips/cip-tx-interface.md (hunk 2 above)
Comment (cmwaters) [reply cips/cip-tx-interface.md:80]: I don't have strong feelings on this. I would lean towards multiplying by 1.1

File: cips/cip-tx-interface.md (hunk 1 above)
Comment (cmwaters) [reply cips/cip-tx-interface.md:9]: lol, good catch

File: cips/cip-tx-interface.md (hunk 3 above)
Comment (cmwaters) [inline cips/cip-tx-interface.md:5]:
```suggestion
discussions-to: https://forum.celestia.org/t/cip-standardised-gas-and-pricing-estimation-interface/1621
```

**Review by rootulp — APPROVED** [review]
CIP content LGTM!

I think the CIP editors need to do the following before it can be merged:
1. assign a CIP number
2. rename the file
3. add it to the table of contents in the README

**Review by jcstein — APPROVED** [review]

**Pull request merged**